device/common/include/traccc/edm/device/triplet_counter.hpp

```diff
@@ -0,0 +1,60 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2021-2022 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+// Project include(s).
+#include "traccc/edm/container.hpp"
+#include "traccc/seeding/detail/doublet.hpp"
+
+namespace traccc::device {
+
+/// Header type for the "triplet counter container"
+///
+/// The header stores summary information about the number of triplets found in
+/// a given geometric bin.
+///
+struct triplet_counter_header {
+
+    /// The total number of middle-bottom spacepoint doublets in a given
+    /// geometric bin.
+    unsigned int m_nMidBot = 0;
+
+    /// The total number of Triplets in a given geometric bin
+    unsigned int m_nTriplets = 0;
+
+    TRACCC_HOST_DEVICE
+    unsigned int get_ref_num() const { return m_nMidBot; }
+
+    void zeros() {
+        m_nMidBot = 0;
+        m_nTriplets = 0;
+    }
```

Comment:
You don't need either of these functions on this class. In the "old" implementation @beomki-yeo was generalising some of the code with the help of these functions. But with the "new" implementation we don't need these anymore.

Reply:
I cannot yet remove the get_ref_num() function as this header is still being used in triplet_finding.sycl. When I make that part with the "new" implementation as well, I'll remove it

Reply:
:+1: